msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AcquireTokenByInteractiveFlowSupplier.java

```diff
@@ -121,6 +132,24 @@ private void openDefaultSystemBrowser(URL url) {
         }
     }
 
+    private static void openDefaultSystemBrowserInMac(URL url){
+        Runtime runtime = Runtime.getRuntime();
+        try {
+            runtime.exec("open " + url);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void openDefaultSystemBrowserInLinux(URL url){
+        Runtime runtime = Runtime.getRuntime();
```

Comment:
I'd implement the full logic from .NET, took a long time to figure those things out: 

- check for sudo and fail
- try all the browser opening tools. xdg-open works on Ubuntu and a few distros, but not all.

Reply:
Done in the latest commit.